config.py

```
@@ -196,33 +196,64 @@ def register_operator(self, ecdsa_private_key, bls_key_pair) -> None:
+        if self.is_sequencer:
+            zlogger.info(
+                "This node is acting as the SEQUENCER. ID: %s",
+                self.NODE["id"],
+            )
+
+    async def find_network_sequencer(self) -> str | None:
+        """Finds the network active sequencer id."""
         sequencing_nodes = self.last_state.sequencing_nodes
         attesting_nodes = self.last_state.attesting_nodes
 
-        total_stake = self.HISTORICAL_NETWORK_STATE[self.NETWORK_STATUS_TAG].total_stake
+        total_stake = self.last_state.total_stake
+
+        from common.auth import create_session
 
-        sequencers_stake: dict[str, Any] = dict.fromkeys(sequencing_nodes, 0)
-        for node_id in attesting_nodes:
-            if node_id == self.NODE["id"]:
-                continue
-            url: str = f"{attesting_nodes[node_id]['socket']}/node/state"
+        async def query_node_state(node_id: str) -> tuple[str, str | None]:
+            """Query a single node's state and return (node_id, sequencer_id)."""
+            url = f"{self.NODES[node_id]['socket']}/node/state"
             try:
-                response = requests.get(url=url, timeout=5).json()
-                if response["data"]["version"] != self.VERSION:
-                    continue
-                sequencer_id = response["data"]["sequencer_id"]
-                if sequencer_id in sequencing_nodes:
-                    sequencers_stake[sequencer_id] += attesting_nodes[node_id]["stake"]
+                async with create_session() as session:
+                    async with session.get(url) as response:
+                        data = await response.json()
+                        if data["data"]["version"] == self.VERSION:
+                            sequencer_id = data["data"]["sequencer_id"]
+                            if sequencer_id in sequencing_nodes:
+                                return node_id, sequencer_id
             except Exception:
```

The caught exception is too broad, which can mask unexpected errors. Additionally, the log message doesn't provide any details about the exception, making it difficult to debug. Consider catching a more specific exception and including the exception details in the log.

---

Fixed